### ISSUE TITLE
Suppressed options

### ISSUE THREAD

**Issue by maxrothman** [Russell91/pythonpy#83]
Several options, including `--split-input` and `-output`, and `--json-input` and `-output` are suppressed from command line help and are not documented. is this intentional? If not, I'd be happy to make a pull request.

**Comment by Russell91** [comment]
Yeah, this is intentional. I simplified the interface soon after launch but
kept the old signals for backwards compatibility reasons. Good eye though.

**Comment by maxrothman** [comment]
Out of curiosity, why were they removed? `--si` especially is quite handy. Here's an example I ran into recently:

I have a file of tab-separated columns:
```
9.561483	9.561483	message 1
114.002299	114.002299	message 2
198.584649	198.584649	yet another message
```
And I want to multiply all of the numbers in the first two columns by 1.25:
```shell
#With --si
py --si '\t' -x '"{:.6f}\t{:.6f}\t{}".format(float(x[0]) * 1.25, float(x[1]) * 1.25, x[2])' <file

#Without --si
py -x '"{:.6f}\t{:.6f}\t{}".format(float(x.split("\t")[0]) * 1.25, float(x.split("\t")[1]) * 1.25, x.split("\t")[2])'
```

Since you can't define variables in `py` programs, and because you can't pass `py` multiple expressions and have I/O piped between them like in Python Power at the Shell or Pyed Piper, doing text manipulation is messy, and `--si` helps alleviate some of that. Perhaps a better fix might be to add support for multiple expressions.

**Comment by maxrothman** [comment]
On another note, I think I've figured out how to get `py` to handle any single complete Python statement, rather than just expressions like it does now, using a combination of `compile(codestring, '', 'single')` and `sys.displayhook`. The result is that code like `x=x.split(); '{} {}'.format(x[0], x[2])` would now work. Would you be interested in a pull request that implements that? The only wrinkle is deciding what to do with statements that have multiple outputs like `1; 2`, but I think a reasonable solution is just to take the last one.

I also think I could get multiple statements with chaining between them working. I only mention it because both features could be used to accomplish the same thing, even though they theoretically have separate utility.

**Comment by maxrothman** [comment]
Any thoughts? Are you interested in getting either or both of those changes?

**Comment by maxrothman** [comment]
Pinging this again. Are you interested either of the changes I described, or should this be closed?

**Comment by rversteegen** [comment]
Well, *I'm* interested in those changes! Could you please put your fork on github?
Also, I think submitting a pull request for changes useful to a subset of users is a good way to allow people to discover them, even if you expect the request will be rejected.

Also, I disagree with hiding useful arguments unless some better replacement to them will be implemented.

Another argument that's useful is --ji (json input), for example it can be used to parse a file containing one number per line from strings into ints/floats. But it's a non-obvious use.

**Comment by maxrothman** [comment]
Given that this is the second comment I've gotten this month on one of my issues in this project (the other being by @vladsaveliev in #82), it seems that there finally might be enough demand for a fork. TBH I haven't thought about these issues or used pythonpy actively in a while, so it might take a while for me to find a weekend to hack something together.

In the meantime, I'd love some feedback on the two solutions I detailed above:
* Expression chaining: a la Pyed Piper or Python Power at the Shell: `py -x 'x.split()' 'f"{x[0]} {x[2]}"'` (yay fstrings!)
  * Is separate arguments the right API for this, rather than internal pipes or something else?
* Full expression parsing: `py -x 'y=x.split(); f"y[0] y[2]"'`
  * Maybe there are other solutions? Llike, a completely different assignment API could be something like `py -x --assign y 'x.split()' 'something' 'something else' 'f"y[0] y[2]"'`

Would we want both of these? If not, which one? What should the API be? What are the use-cases for these features?

I'll do some more thinking about these questions myself, but I'd really like to gather more viewpoints as well.

**Comment by vladsaveliev** [comment]
Confirm that I'm interested in it, and from the first glance I like the full expression parsing suggestion a lot. Ability to assign variables can have so many applications in the long run.

**Comment by Russell91** [comment]
@maxrothman sorry for the slow response. Feel free to modify the help command to add back support for --si, --so, --ji, and --jo.

**Comment by maxrothman** [comment]
Since this was resolved in 71c81c8596c709b1ad3785539349144747f66a37, I'm closing this and we can continue discussion of future enhancements in https://github.com/Russell91/pythonpy/issues/86.
